js_modules/dagster-ui/packages/ui-core/src/ui/__tests__/formatDuration.test.ts

```diff
@@ -0,0 +1,208 @@
+        {value: 2, unit: 'minutes'},
+      ]);
+      expect(formatDuration(93784000, {significantDigits: 2})).toEqual([
+        {value: 26, unit: 'hours'},
+        {value: 3, unit: 'minutes'},
+      ]);
+    });
+  });
+
+  describe('seconds input mode', () => {
+    it('should convert seconds to appropriate units', () => {
+      expect(formatDuration(1, {unit: 'seconds'})).toEqual([{value: 1, unit: 'second'}]);
+      expect(formatDuration(60, {unit: 'seconds'})).toEqual([{value: 1, unit: 'minute'}]);
+      expect(formatDuration(3600, {unit: 'seconds'})).toEqual([{value: 1, unit: 'hour'}]);
+      expect(formatDuration(86400, {unit: 'seconds'})).toEqual([{value: 1, unit: 'day'}]);
+    });
+
+    it('should work with two significant digits', () => {
+      expect(formatDuration(1000, {unit: 'seconds', significantDigits: 2})).toEqual([
+        {value: 16, unit: 'minutes'},
+        {value: 40, unit: 'seconds'},
+      ]);
+      expect(formatDuration(3665, {unit: 'seconds', significantDigits: 2})).toEqual([
+        {value: 1, unit: 'hour'},
+        {value: 1, unit: 'minute'},
+      ]);
+      expect(formatDuration(90000, {unit: 'seconds', significantDigits: 2})).toEqual([
+        {value: 25, unit: 'hours'},
+      ]);
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should handle very small values', () => {
+      expect(formatDuration(0.5)).toEqual([{value: 0, unit: 'milliseconds'}]);
+      expect(formatDuration(0.9)).toEqual([{value: 0, unit: 'milliseconds'}]);
+    });
+
+    it('should handle very large values', () => {
+      expect(formatDuration(Number.MAX_SAFE_INTEGER)).toEqual([{value: 285616, unit: 'years'}]);
+    });
+
+    it('should handle fractional milliseconds by flooring', () => {
+      expect(formatDuration(1500.7)).toEqual([{value: 1, unit: 'second'}]);
+      expect(formatDuration(1500.7, {significantDigits: 2})).toEqual([
+        {value: 1, unit: 'second'},
+        {value: 500, unit: 'milliseconds'},
+      ]);
+    });
+
+    it('should maintain singular/plural consistency', () => {
+      expect(formatDuration(1000)).toEqual([{value: 1, unit: 'second'}]);
+      expect(formatDuration(2000)).toEqual([{value: 2, unit: 'seconds'}]);
+      expect(formatDuration(60000)).toEqual([{value: 1, unit: 'minute'}]);
+      expect(formatDuration(120000)).toEqual([{value: 2, unit: 'minutes'}]);
+      expect(formatDuration(3600000)).toEqual([{value: 1, unit: 'hour'}]);
+      expect(formatDuration(7200000)).toEqual([{value: 2, unit: 'hours'}]);
+    });
+  });
+
+  describe('real-world examples', () => {
+    it('should format common durations correctly', () => {
+      // API response time
+      expect(formatDuration(247)).toEqual([{value: 247, unit: 'milliseconds'}]);
+
+      // Video duration
+      expect(formatDuration(215000)).toEqual([{value: 3, unit: 'minutes'}]);
+      expect(formatDuration(215000, {significantDigits: 2})).toEqual([
+        {value: 3, unit: 'minutes'},
```

Comment:
I'm having a hard time picturing how this will be displayed in the UI. Will this be rendered as "3 minutes, 35 seconds" in the UI? That seems like it'll be too long to fit nicely in the graphs. I was picturing something more like "3.54 minutes" in that case

Reply:
Hmm, I briefly thought of fractional units but wasn't sure if they would be intuitive. Instead we're just rounding after a certain threshold. eg. for minutes we go as far as 120 minutes, but after that we start rounding to hours. Josh thought that for some of the cards we could show 2 units and others 1. @braunjj wdyt?